src/server/app.js

```diff
@@ -28,14 +33,85 @@ if (process.env.NODE_ENV === 'production-notugly') {
   )
 }
 
-const staticHandler = express.static(path.resolve(__dirname, '../../', 'build'))
+const hbs = exphbs.create({
+  // Specify helpers which are only registered on this instance.
+  extname: '.hbs',
+  helpers: {
+    foo: function () {
+      return 'FOO!'
+    },
+    bar: function () {
+      return 'BAR!'
+    }
+  }
+})
 
-app.use(staticHandler)
-app.use('/embed', staticHandler)
-app.get('/embed/:id', videoRoute)
-app.get('/oembed', oembedRoute)
-app.get('*', (req, res) => {
-  res.sendFile(path.resolve(__dirname, '../../', 'build', 'index.html'))
+app.engine('.hbs', hbs.engine)
+app.set('view engine', 'hbs')
+app.set('views', 'src/views')
+app.use(express.static(path.resolve(__dirname, '../../', 'build')))
+
+app.get('/embed/:id', async function (req, res, next) {
+  const { id } = req.params
+  const video = await paratii.core.vids.get(id)
+  let meta = ''
+
+  if (video !== null) {
+    meta = '<link rel="stylesheet" type="text/css" href="/embed/index.css">'
+    meta = routeHelper.openGraphHead(meta, video)
+    meta = routeHelper.twitterCardHead(meta, video)
+  } else {
+    meta = routeHelper.notFoundVideo(meta)
+  }
+  res.render('index', {
+    helpers: {
+      meta: function () {
+        return meta
+      },
+      script: function () {
+        return '<script type="text/javascript" src="/embed/bundle.js"></script>'
+      }
+    }
+  })
 })
 
+app.get('/play/:id', async function (req, res, next) {
+  const { id } = req.params
+  const video = await paratii.core.vids.get(id)
+  let meta = ''
+
+  if (video !== null) {
+    meta = routeHelper.openGraphHead(meta, video)
+    meta = routeHelper.twitterCardHead(meta, video)
+  } else {
+    meta = routeHelper.notFoundVideo(meta)
+  }
+  res.render('index', {
+    helpers: {
+      meta: function () {
+        return meta
+      },
+      script: function () {
+        return '<script type="text/javascript" src="/bundle.js"></script>'
+      }
+    }
```

Comment:
It looks like there is duplicate logic with this and the route above. Why the differences? Can this be DRY'd up?

Reply:
we can put this inside a conditional statement (embed or not) inside an handlebars template. What do you think?